sound/soc/sof/pcm.c

```diff
@@ -288,9 +288,41 @@ static int sof_pcm_hw_free(struct snd_soc_component *component,
 	struct snd_sof_pcm *spcm;
 	int ret, err = 0;
 
-	/* nothing to do for BE */
-	if (rtd->dai_link->no_pcm)
+	if (rtd->dai_link->no_pcm) {
```

Comment:
It is worthy to have a comment to explain why we need to do this..

Reply:
hmm...do you think we need a separate function for this?

Reply:
forget it....the function is in component driver...

Reply:
@brentlu if here is late enough? it is in component driver .hw_free() which happen still before dai->driver->ops->hw_free()?

Reply:
Codecs like cs42l42 switches PLL in the mute_stream() callback so it should be enough. I'm still waiting their feedback.